### PR TITLE
test: add tests for environment variables and runtime config overrides

### LINKED ISSUE #1150: Add tests for process.env support
Cloudflare preset now supports global env. We shall add tests to verify `process.env.FOO_BAR` dynamic access works as expected across presets (and ignore the one's still have known issues)

### PR DESCRIPTION
**Pull request by Hebilicious**
### 🔗 Linked issue

Extracted from https://github.com/unjs/nitro/pull/1318

closes #1150

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

While working on https://github.com/unjs/nitro/pull/1318 , I realised it was better to extract the tests from this PR into a standalone PR, as they can now be used to test `useRuntimeConfig` with every presets. 
I noticed a bug in the `cloudflare` preset as well that I included here, however I can include it in another PR if that's better.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/entries/cloudflare.ts [hunk 1]
```diff
@@ -29,6 +29,8 @@ async function handleEvent(event: FetchEvent) {
     body = Buffer.from(await event.request.arrayBuffer());
   }
 
+  globalThis.__env__ = globalThis;
```
Comment (Hebilicious) [inline src/runtime/entries/cloudflare.ts:32]: Should this be moved to a separate issue/PR ?

Comment (pi0) [reply src/runtime/entries/cloudflare.ts:32]: Yes, please. Also i'm wondering why this line is not working in unenv:

https://github.com/unjs/unenv/blob/f4cba4ec85becd2e7f4f4d995c46ee34f8ab49dc/src/runtime/node/process/_process.ts#L163

(by doing above, setting something to `process.env` in cf preset causes updating globalThis)

Comment (Hebilicious) [reply src/runtime/entries/cloudflare.ts]: Disabled the tests for now. I'm not entirely sure why it's happening, it appears I'm unable to reproduce the bug outside of the tests 🤔

Comment (Hebilicious) [reply src/runtime/entries/cloudflare.ts]: It was because of a wrong Miniflare configuration 🤦🏽 Fixed now

**Review by pi0 — APPROVED** [review]
Thanks!

**Pull request merged**